packages/ssm-cache/index.ts

```diff
@@ -46,31 +55,75 @@ export class SsmParameterNotFound extends Error {
   }
 }
 
-export const getSsmParameter = (name: string): string => {
-  if (!Object.prototype.hasOwnProperty.call(ssmCache.values, name)) {
-    throw new SsmParameterNotFound(`SSM parameter ${name} not found in cache`);
+export const hasParameterExpired = (parameter: SsmCacheParameter | undefined) => {
+  return !!(parameter?.expiryDate && new Date() > parameter.expiryDate);
+};
+
+export const hasCacheExpired = () => !!(ssmCache.expiryDate && new Date() > ssmCache.expiryDate);
+
+export const isConfigNotEmpty = () => !!Object.keys(ssmCache.values).length;
+
+export const setCacheDurationMilliseconds = (cacheDurationMilliseconds: number) => {
+  ssmCache.cacheDurationMilliseconds = cacheDurationMilliseconds;
+};
+
+export const parameterExistsInCache = (name: string) =>
+  Object.prototype.hasOwnProperty.call(ssmCache.values, name) && ssmCache.values[name];
+
+export const getSsmClient = () => {
+  if (!ssm) {
+    ssm = new SSM(ssmConfig);
   }
 
-  return ssmCache.values[name] || '';
+  return ssm;
 };
 
 export const addToCache = (regex: RegExp | undefined, { Name, Value }: SSM.Parameter) => {
   if (!Name) return;
   if (regex && !regex.test(Name)) return;
 
-  ssmCache.values[Name] = Value;
+  ssmCache.values[Name] = {
+    value: Value || '',
+    expiryDate: new Date(Date.now() + ssmCache.cacheDurationMilliseconds),
+  };
 };
 
-export const hasCacheExpired = () => !!(ssmCache.expiryDate && new Date() > ssmCache.expiryDate);
+export const loadParameter = async (name: string) => {
+  const params: SSM.GetParameterRequest = {
+    Name: name,
+    WithDecryption: true,
+  };
 
-export const isConfigNotEmpty = () => !!Object.keys(ssmCache.values).length;
+  const { Parameter } = await getSsmClient()
+    .getParameter(params)
+    .promise();
 
-export const getSsmClient = () => {
-  if (!ssm) {
-    ssm = new SSM(ssmConfig);
+  if (!Parameter?.Name) {
+    return;
   }
 
-  return ssm;
+  addToCache(undefined, Parameter);
+};
+
+export const getSsmParameter = async (name: string): Promise<string> => {
+  // If the cache doesn't have the requested parameter or if it is expired, load it
+
+  if (!parameterExistsInCache(name) || hasParameterExpired(ssmCache.values[name])) {
+    await loadParameter(name);
+  }
+
+  // If the cache still doesn't have the requested parameter, throw an error
+  if (!parameterExistsInCache(name)) {
+    throw new SsmParameterNotFound(`Parameter ${name} not found`);
+  }
+
+  return ssmCache?.values[name]?.value || '';
```

Comment:
Would ssmCache ever not be defined? We don't check for it in `parameterExistsInCache`